lib/scoped_search/query_builder.rb

```diff
@@ -485,6 +489,16 @@ def to_sql(builder, definition, &block)
             raise ScopedSearch::QueryNotSupported, "Don't know how to handle this operator node: #{operator.inspect} with #{children.inspect}!"
           end
         end
+
+        private
+
+        def validate_value(field, value)
+          validator = field.validator
+          if field.validator && field.only_explicit
+            valid = validator.call(value)
```

Comment:
As an alternative, we could require a validator to `raise ScopedSearch::ValidationFailed, "my message"`. That might be a little bit less subtle, but has a smaller chance of being accidentally used the wrong way.

I still have scars from Rails's `before_filter`s stopping further request processing if you'd return something false-ish from them. :)